### ISSUE TITLE
MA0095 behaves differently to CA1067 with CRTP

### ISSUE THREAD

**Issue by daverant** [meziantou/Meziantou.Analyzer#917]
### Version of the Meziantou.Analyzer NuGet package

2.0.239

### Rule Identifier

MA0095

### Target Framework

net9.0

### C# Language version

C# 13

### Description

Unsure if the difference in behaviour between MA0085 and CA1067 is expected, but MA0095 fires when using [CRTP](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern) and CA1067 does not. I thought it was worth reporting just in-case. Thanks!

### Reproduction Steps

Minimal code:

```c#
public abstract class Crtp<T> : IEquatable<T> where T : Crtp<T>
{
    public bool Equals(T? other) => true;
}

public sealed class TriggersMA0095Only : Crtp<TriggersMA0095Only>;

public sealed class TriggersMA0095AndCA1067 : IEquatable<TriggersMA0095AndCA1067>
{
    public bool Equals(TriggersMA0095AndCA1067? other) => true;
}
```

### Other information

Updating Crtp<T> to the following has no change on analyzers that are triggered:
```c#
public abstract class Crtp<T> : IEquatable<T> where T : Crtp<T>
{
    public bool Equals(T? other) => true;
    public override bool Equals(object? obj) => true;
    public override int GetHashCode() => 0;
}
```

**Comment by meziantou** [comment]
I've updated the rule to report only on a type that directly implement the interface.
